test/e2e/operator/deployment_api.go

```diff
@@ -721,3 +756,35 @@ func findOperatorDeployment(f *framework.Framework) (*appsv1.Deployment, error)
 
 	return dep, nil
 }
+
+// deleteOperatorDeployment removes the operator deployment object
+func deleteOperatorDeployment(f *framework.Framework) error {
+	dep, err := findOperatorDeployment(f)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	framework.Logf("Deleting the operator '%s/%s' running", dep.Namespace, dep.Name)
+	if err = f.ClientSet.AppsV1().Deployments(dep.Namespace).Delete(dep.Name, nil); err != nil {
```

Comment:
It'll be simpler to just repeat the `Delete` call until you get a `NotFound` error. As an added bonus, the code becomes more robust against temporary delete failures.

Reply:
Done. Instead of deleting deployment, making its replication count to zero and waiting till the operator pod gets deleted in a repeated loop.